### PR TITLE
Fix Rust support after recent changes

### PR DESCRIPTION
**Pull request by nazar-pc**
First commit restores code unnecessarily removed in #893 without which Rust can't send anything to worker.

Second commit fixes Rust support after #893, there I had to change how requests and notifications look like a bit, as the result we have less code and it is a bit cleaner.

Third commit updates dump data structure and tests according to changes in #894.

Last commit bumps versions, dependencies and updates changelog so we can publish it.

### CODE REVIEW

**Review by ibc — APPROVED** [review]
Thanks.

File: worker/src/Channel/ChannelSocket.cpp [hunk 1]
```diff
@@ -191,6 +191,43 @@ namespace Channel
 		auto free = this->channelReadFn(
 		  &message, &messageLen, &messageCtx, this->uvReadHandle, this->channelReadCtx);
 
+		if (free)
```
Comment (ibc) [inline worker/src/Channel/ChannelSocket.cpp:198]: Please add some inline doc here telling what it does.

Comment (ibc) [reply worker/src/Channel/ChannelSocket.cpp:198]: Also, is this code executed when using Node? If so, could it be disabled via some define passed when building the worker?

Comment (nazar-pc) [reply worker/src/Channel/ChannelSocket.cpp:198]: `ChannelSocket::CallbackRead` is not called by Node, this is just for direct communication with worker when we can make a function call. Will add some comments.

**Review by jmillan — APPROVED** [review]

**Comment by ibc** [comment]
Any chance to have this merged tomorrow morning? I'd like to make a release for Node and I assume we need another one for Rust.

**Comment by nazar-pc** [comment]
It is just the comments that are left. I'll try to write some within next few hours.

File: worker/src/Channel/ChannelSocket.cpp (hunk 1 above)
Comment (nazar-pc) [reply worker/src/Channel/ChannelSocket.cpp:198]: Do added comments make sense?

Comment (ibc) [reply worker/src/Channel/ChannelSocket.cpp:198]: Yes, but I miss some comments telling that the block is only used in Rust due to how message channel works in Rust (within some process) compared to how it works in Node (inter-process comm).

Comment (nazar-pc) [reply worker/src/Channel/ChannelSocket.cpp:198]: Hm, if you go to the place from which this function is called, it makes total sense

Comment (nazar-pc) [reply worker/src/Channel/ChannelSocket.cpp:198]: I described the whole workflow in https://github.com/versatica/mediasoup/pull/666 that introduced this code

Comment (ibc) [reply worker/src/Channel/ChannelSocket.cpp:198]: Merge please 😀

**Comment by ibc** [comment]
I'll merge. Thanks for this.

**Comment by ibc** [comment]
3.6.10 released

**Comment by nazar-pc** [comment]
Rust 0.11.0 is out too

**Pull request merged**